src/Castle.Windsor.Extensions.MsDependencyInjection/NetCoreRootScopeAccessor.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2004-2020 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Windsor.Extensions.MsDependencyInjection
+{
+    using System;
+    
+    using Castle.MicroKernel.Context;
+    using Castle.MicroKernel.Lifestyle.Scoped;
+
+    internal class NetCoreRootScopeAccessor : IScopeAccessor
+    {
+        public ILifetimeScope GetScope(CreationContext context)
+        {
+            if(NetCoreScope.Current == null)
+            {
+                throw new InvalidOperationException("No scope");
+            }
+
+            if(NetCoreRootScope.Current.RootScope == null)
+            {
+                throw new InvalidOperationException("No root scope");
+            }
+
+            return NetCoreRootScope.Current.RootScope;     
+        }
+
+        public void Dispose()
```

Comment:
What is the dispose of normally used for in the IScopeAccessor, is something missing here? Just i question, i am unsure...

Reply:
It's required by interface `IScopeAccessor`. I am not doing anything sicne the actual scope is disposed based on .NET Core scope